Library/Homebrew/cli/named_args.rb

```diff
@@ -73,6 +86,18 @@ def to_resolved_formulae_to_casks
         end
       end
 
+      # Convert named arguments to `Tap`, `Formula` or `Cask` objects.
+      # If both a formula and cask exist with the same name, returns the
+      # formula and prints a warning unless `only` is specified.
+      def to_objects(only: nil)
+        @to_objects ||= {}
+        @to_objects[only] ||= downcased_unique_named.flat_map do |name|
+          next Tap.fetch(name) if only == :tap || (only.nil? && name.count("/") == 1 && !name.start_with?("./", "/"))
```

Comment:
Could this get split into a few conditionals or variables? It's a bit hard to follow as a post-if.